sky/clouds/oci.py

```diff
@@ -0,0 +1,387 @@
+                                    region: Optional[str] = None,
+                                    zone: Optional[str] = None) -> float:
+        del accelerators, use_spot, region, zone  # unused
+        return 0.0
+
+    def get_egress_cost(self, num_gigabytes: float) -> float:
+        """
+        https://www.oracle.com/cis/cloud/networking/pricing/
+        """
+        # Free for first 10T (per month)
+        if num_gigabytes <= 10 * 1024: 
+            return 0.0 
+        
+        # We need to calculate the egress cost by region.
+        # Fortunately, most of time, this cost looks not a 
+        # big deal comparing to the price of GPU instances.
+        # # Calculate cost for over 10T (per month)
+        # logger.debug(f"* get_egress_cost. region {region}")
+        # if region in REGIONS_NorthAmerica_Europe_UK:
+        #     return (num_gigabytes - 10 * 1024) * 0.0085
+        # elif region in REGIONS_JAPAC_SouthAmerica:
+        #     return (num_gigabytes - 10 * 1024) * 0.025
+        # elif region in REGIONS_MidEast_Africa:
+        #     return (num_gigabytes - 10 * 1024) * 0.05
+        # else:
+        #     logger.debug(f"* ! Region {region} is not listed for cost calc!")
+        #     return 0.0
+        return (num_gigabytes - 10 * 1024) * 0.0085
+
+    def __repr__(self):
+        return 'oci'
+
+    def is_same_cloud(self, other: clouds.Cloud) -> bool:
+        # Returns true if the two clouds are the same cloud type.
+        return isinstance(other, OCI)
+    
+
+    @classmethod
+    def get_default_instance_type(
+            cls,
+            cpus: Optional[str] = None,
+            memory: Optional[str] = None,
+            disk_tier: Optional[str] = None) -> Optional[str]:
+        return service_catalog.get_default_instance_type(cpus=cpus,
+                                                         memory=memory,
+                                                         disk_tier=disk_tier,
+                                                         clouds='oci')
+
+    @classmethod
+    def get_accelerators_from_instance_type(
+        cls,
+        instance_type: str,
+    ) -> Optional[Dict[str, int]]:
+        return service_catalog.get_accelerators_from_instance_type(
+            instance_type, clouds='oci')
+
+
+    @classmethod
+    def get_vcpus_from_instance_type(
+        cls,
+        instance_type: str,
+    ) -> Optional[float]:
+        return service_catalog.get_vcpus_from_instance_type(instance_type,
+                                                            clouds='oci')
+
+
+    @classmethod
+    def get_zone_shell_cmd(cls) -> Optional[str]:
+        return None
+
+
+    def make_deploy_resources_variables(
+            self, resources: 'resources_lib.Resources',
+            region: Optional['clouds.Region'],
+            zones: Optional[List['clouds.Zone']]) -> Dict[str, Optional[str]]:
+        del zones
+        if region is None:
+            region = self._get_default_region()
```

Comment:
We can `assert region is not None` here, as our `cloud_vm_ray_backend` should guarantee that the region is not None.

Reply:
Let me perform a test first.